sdk/include/opentelemetry/sdk/trace/samplers/always_on.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/sampler.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+/**
+ * The always on sampler is a default sampler which always return Decision::RECORD_AND_SAMPLE
+ */
+class AlwaysOnSampler : public Sampler
+{
+public:
+  /**
+   * @return Always return Decision RECORD_AND_SAMPLE
+   */
+  SamplingResult ShouldSample(const SpanContext * /*parent_context*/,
+                              trace_api::TraceId /*trace_id*/,
+                              nostd::string_view /*name*/,
+                              trace_api::SpanKind /*span_kind*/,
+                              const trace_api::KeyValueIterable & /*attributes*/) noexcept override
+  {
+    return {Decision::RECORD_AND_SAMPLE, nullptr};
+  }
+
+  /**
+   * @return Description MUST be AlwaysOnSampler
+   */
+  std::string GetDescription() const noexcept override { return "AlwaysOnSampler"; }
```

Comment:
If description is always expected to be a string literal, we could return a string_view instead of a std::string.

Reply:
Agreed. a `string_view` can be more efficient. I need to open another PR for this issue, since we need to change the data type in multiple header files.